### PR TITLE
nss-systemd: don't synthesize root/nobody when iterating

### PR DESCRIPTION
**Pull request by poettering**
Fixes: #15160

### CODE REVIEW

File: src/nss-systemd/nss-systemd.c [hunk 1]
```diff
@@ -329,7 +334,7 @@ enum nss_status _nss_systemd_setgrent(int stayopen) {
         getgrent_data.iterator = userdb_iterator_free(getgrent_data.iterator);
         getpwent_data.by_membership = false;
 
-        r = groupdb_all(nss_glue_userdb_flags(), &getgrent_data.iterator);
+        r = groupdb_all(nss_glue_userdb_flags() | USERDB_DONT_SYNTHESIZE, &getgrent_data.iterator);
```
Comment (jlebon) [inline src/nss-systemd/nss-systemd.c:338]: Maybe a comment here to look at the comment block in `_nss_systemd_setpwent`?

Comment (poettering) [reply src/nss-systemd/nss-systemd.c:338]: sure why not

**Comment by poettering** [comment]
pushed new version with the comment @jlebon suggested added. no other changes.

**Comment by keszybz** [comment]
bionic-amd64 timed out in TEST-27-STDOUTFILE, looks unrelated.

**Pull request merged**